modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/display/context/ViewTreeManagementToolbarDisplayContext.java

```diff
@@ -506,15 +506,30 @@ else if (className.equals(User.class.getName())) {
 					navigationStatus, null));
 		}
 		else {
-			searchContainer.setResultsAndTotal(
-				() -> OrganizationLocalServiceUtil.getOrganizationsAndUsers(
+			List<Object> organizationsAndUsers = new ArrayList<>();
+
+			organizationsAndUsers.addAll(
+				OrganizationLocalServiceUtil.getOrganizations(
 					_themeDisplay.getCompanyId(),
-					_organization.getOrganizationId(), navigationStatus,
-					searchContainer.getStart(), searchContainer.getEnd(),
-					searchContainer.getOrderByComparator()),
+					_organization.getOrganizationId()));
+
+			organizationsAndUsers.addAll(
+				UserLocalServiceUtil.getOrganizationUsers(
+					_organization.getOrganizationId()));
+
+			int organizationsAndUsersCount =
 				OrganizationLocalServiceUtil.getOrganizationsAndUsersCount(
 					_themeDisplay.getCompanyId(),
-					_organization.getOrganizationId(), navigationStatus));
+					_organization.getOrganizationId(), navigationStatus);
+
+			int end =
+				(organizationsAndUsersCount < searchContainer.getEnd()) ?
+					organizationsAndUsersCount : searchContainer.getEnd();
+
+			searchContainer.setResultsAndTotal(
```

Comment:
I think this is not the best way to implement it because if you have a lot of records in the database you extract them all to filter the list you are showing. The best way should be to pass the start/end parameters to the query